src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -63,23 +64,26 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
 
   override type M = ErgoStateContext
 
-  def sigmaPreHeader: special.sigma.PreHeader = PreHeader.toSigma(lastHeaders.head)
+  override def sigmaPreHeader: special.sigma.PreHeader =
+    PreHeader.toSigma(lastHeaders.headOption.getOrElse(PreHeader.fake))
 
-  def sigmaLastHeaders: Coll[special.sigma.Header] = new CollOverArray(lastHeaders.tail.map(h => Header.toSigma(h)).toArray)
-
-  // todo remove from ErgoLikeContext and from ErgoStateContext
-  def lastBlockMinerPk: Array[Byte] = sigmaPreHeader.minerPk.getEncoded.toArray
+  override def sigmaLastHeaders: Coll[special.sigma.Header] = new CollOverArray(lastHeaders.drop(1).map(h => Header.toSigma(h)).toArray)
```

Comment:
why `tail` is replaced with `drop(1)`?

Reply:
To handle empty `lastHeaders` (without throwing).